src/ui/page/channel/view.jsx

```diff
@@ -48,44 +99,179 @@ function ChannelPage(props: Props) {
     history.push(`${url}${search}`);
   };
 
+  const handleBidChange = (bid: number) => {
+    const { balance, amount } = props;
+    const totalAvailableBidAmount = parseFloat(amount) + parseFloat(balance);
+    setParams({ ...params, amount: bid });
+    setBidError('');
+    if (bid <= 0.0 || isNaN(bid)) {
+      setBidError(__('Deposit cannot be 0'));
+    } else if (totalAvailableBidAmount === bid) {
+      setBidError(__('Please decrease your deposit to account for transaction fees'));
+    } else if (totalAvailableBidAmount < bid) {
+      setBidError(__('Deposit cannot be higher than your balance'));
+    } else if (bid <= MINIMUM_PUBLISH_BID) {
+      setBidError(__('Your deposit must be higher'));
+    }
+  };
```

Comment:
We use this same logic in the publish form. Lets pull it into it's own function.

It can accept `bid, balance, amount` and return an error if it's bad. If there is no error, update the form, if there is an error, setError

```
const error = checkBidStuff(amount, balance, bid)
if (error) { 
  setBidError(error)
} else {
  setParams()
}

Reply:
DIdn't do this yet. 
I'm thinking checkBidStuff(amount, balance, bid, errorFn, updateFn) would be the signature? Or do I misunderstand you?